### PR TITLE
[module] driver for the TeraRanger One range finder

### PR DESCRIPTION
**Pull request by gautierhattenberger**

### CODE REVIEW

File: conf/modules/teraranger_one.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE module SYSTEM "module.dtd">
+
+<module name="teraranger_one" dir="range_finder">
+  <doc>
+    <description>Driver for the TeraRanger One range finder (I2C)</description>
+    <configure name="TERARANGER_ONE_I2C_DEV" value="i2cX" description="Select I2C bus"/>
+    <define name="TERARANGER_ONE_I2C_ADDR" value="0x30" description="Select an alternate I2C address"/>
+    <define name="TERARANGER_ONE_OFFSET" value="0.0" description="Add an offset to the measurments" unit="m"/>
+    <define name="USE_TERARANGER_ONE_AGL" value="TRUE|FALSE" description="Send data as AGL measurments to the system"/>
```
Comment (flixr) [inline conf/modules/teraranger_one.xml]: Does it even make sense to be able to turn this off? If yes, should probably be mentioned that it is TRUE by default.

File: conf/modules/teraranger_one.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE module SYSTEM "module.dtd">
+
+<module name="teraranger_one" dir="range_finder">
+  <doc>
+    <description>Driver for the TeraRanger One range finder (I2C)</description>
+    <configure name="TERARANGER_ONE_I2C_DEV" value="i2cX" description="Select I2C bus"/>
```
Comment (flixr) [inline conf/modules/teraranger_one.xml]: Should probably mention that i2c2 is the default...

File: sw/airborne/modules/range_finder/teraranger_one.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+#ifndef TERARANGER_ONE_H
+#define TERARANGER_ONE_H
+
+#include "std.h"
+
+struct TeraRanger {
+  uint16_t raw; ///< raw distance in mm
+  float dist;   ///< scaled distance in m
+  float offset; ///< offset in m
+  bool data_available;
```
Comment (flixr) [inline sw/airborne/modules/range_finder/teraranger_one.h]: doesn't seem to be read anywhere, is this still supposed to be used somewhere?

File: conf/modules/teraranger_one.xml (hunk 1 above)
Comment (gautierhattenberger) [inline conf/modules/teraranger_one.xml]: This is for the case where you don't want to use the sensor to detect ground but something else, like obstacle in front of you. In this case you don't want to send the ABI AGL message.

File: sw/airborne/modules/range_finder/teraranger_one.h (hunk 3 above)
Comment (gautierhattenberger) [inline sw/airborne/modules/range_finder/teraranger_one.h]: This is used when you don't use the sensor through ABI message but with direct reading (see remark above).

File: conf/modules/teraranger_one.xml (hunk 2 above)
Comment (gautierhattenberger) [inline conf/modules/teraranger_one.xml]: done

**Pull request merged**